diff -Naur graphlcd-base-9999.orig/Make.config graphlcd-base-9999/Make.config
--- graphlcd-base-9999.orig/Make.config	2012-10-15 23:25:15.000000000 +0200
+++ graphlcd-base-9999/Make.config	2012-10-15 23:30:07.399792704 +0200
@@ -64,3 +64,6 @@
 # uncomment this variable if you want to enable the experimental AX 206 based digital photo frame driver
 # Read DRIVER.ax206dpf before use!
 #HAVE_AX206DPF_EXPERIMENTAL=1
+
+# comment this if you do not want to have the binaries pre-stripped on 'make install'
+PRESTRIP = -s
diff -Naur graphlcd-base-9999.orig/tools/convpic/Makefile graphlcd-base-9999/tools/convpic/Makefile
--- graphlcd-base-9999.orig/tools/convpic/Makefile	2012-10-15 23:25:15.000000000 +0200
+++ graphlcd-base-9999/tools/convpic/Makefile	2012-10-15 23:32:04.949323134 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -o root -g root $(PRESTRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-9999.orig/tools/crtfont/Makefile graphlcd-base-9999/tools/crtfont/Makefile
--- graphlcd-base-9999.orig/tools/crtfont/Makefile	2012-10-15 23:25:15.000000000 +0200
+++ graphlcd-base-9999/tools/crtfont/Makefile	2012-10-15 23:31:47.590540148 +0200
@@ -35,7 +35,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -o root -g root $(PRESTRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-9999.orig/tools/genfont/Makefile graphlcd-base-9999/tools/genfont/Makefile
--- graphlcd-base-9999.orig/tools/genfont/Makefile	2012-10-15 23:25:15.000000000 +0200
+++ graphlcd-base-9999/tools/genfont/Makefile	2012-10-15 23:31:30.651751911 +0200
@@ -38,7 +38,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -o root -g root $(PRESTRIP) $(PRGNAME) $(BINDIR)
 	
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-9999.orig/tools/lcdtestpattern/Makefile graphlcd-base-9999/tools/lcdtestpattern/Makefile
--- graphlcd-base-9999.orig/tools/lcdtestpattern/Makefile	2012-10-15 23:25:15.000000000 +0200
+++ graphlcd-base-9999/tools/lcdtestpattern/Makefile	2012-10-15 23:31:13.166970501 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -o root -g root $(PRESTRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-9999.orig/tools/showpic/Makefile graphlcd-base-9999/tools/showpic/Makefile
--- graphlcd-base-9999.orig/tools/showpic/Makefile	2012-10-15 23:25:15.000000000 +0200
+++ graphlcd-base-9999/tools/showpic/Makefile	2012-10-15 23:30:54.474204194 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -o root -g root $(PRESTRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-9999.orig/tools/showtext/Makefile graphlcd-base-9999/tools/showtext/Makefile
--- graphlcd-base-9999.orig/tools/showtext/Makefile	2012-10-15 23:25:15.000000000 +0200
+++ graphlcd-base-9999/tools/showtext/Makefile	2012-10-15 23:30:33.864461849 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -o root -g root $(PRESTRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
