--- xbmc-9999.ebuild	2014-08-12 07:01:13.000000000 +0200
+++ xbmc-13.2.ebuild	2014-08-21 17:14:10.000000000 +0200
@@ -1,6 +1,6 @@
 # Copyright 1999-2014 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
-# $Header: /var/cvsroot/gentoo-x86/media-tv/xbmc/xbmc-9999.ebuild,v 1.164 2014/08/12 04:47:55 vapier Exp $
+# $Header: /var/cvsroot/gentoo-x86/media-tv/xbmc/xbmc-13.2.ebuild,v 1.1 2014/08/21 15:14:10 vapier Exp $
 
 EAPI="5"
 
@@ -11,7 +11,7 @@
 
 inherit eutils python-single-r1 multiprocessing autotools
 
-CODENAME="Helix"
+CODENAME="Gotham"
 case ${PV} in
 9999)
 	EGIT_REPO_URI="git://github.com/xbmc/xbmc.git"
@@ -33,10 +33,7 @@
 		http://mirrors.xbmc.org/releases/source/${MY_P}-generated-addons.tar.xz"
 	KEYWORDS="~amd64 ~x86"
 
-	S=${WORKDIR}/${PN}-
-	[[ ${PV} == *_p* ]] \
-		&& S+=${PV/_p/-${CODENAME}_r} \
-		|| S+=${MY_PV}
+	S=${WORKDIR}/${MY_P}-${CODENAME}
 	;;
 esac
 
@@ -45,7 +42,7 @@
 
 LICENSE="GPL-2"
 SLOT="0"
-IUSE="airplay alsa altivec avahi bluetooth bluray caps cec css debug +fishbmc gles goom java joystick midi mysql nfs +opengl profile +projectm pulseaudio pvr +rsxs rtmp +samba +sdl sse sse2 sftp test udisks upnp upower +usb vaapi vdpau webserver +X +xrandr"
+IUSE="airplay alsa altivec avahi bluetooth bluray caps cec css debug +fishbmc gles goom java joystick midi mysql nfs +opengl profile +projectm pulseaudio pvr +rsxs rtmp +samba +sdl sse sse2 sftp udisks upnp upower +usb vaapi vdpau webserver +X +xrandr"
 REQUIRED_USE="
 	pvr? ( mysql )
 	rsxs? ( X )
@@ -145,8 +142,7 @@
 	X? ( x11-proto/xineramaproto )
 	dev-util/cmake
 	x86? ( dev-lang/nasm )
-	java? ( virtual/jre )
-	test? ( dev-cpp/gtest )"
+	java? ( virtual/jre )"
 # Force java for latest git version to avoid having to hand maintain the
 # generated addons package.  #488118
 [[ ${PV} == "9999" ]] && DEPEND+=" virtual/jre"
@@ -257,7 +253,6 @@
 		$(use_enable sdl) \
 		$(use_enable sftp ssh) \
 		$(use_enable usb libusb) \
-		$(use_enable test gtest) \
 		$(use_enable upnp) \
 		$(use_enable vaapi) \
 		$(use_enable vdpau) \
