diff -Naur yacoto-0.4_orig/yac-menurecordings.c yacoto-0.4/yac-menurecordings.c
--- yacoto-0.4_orig/yac-menurecordings.c	2008-12-18 23:00:29.000000000 +0100
+++ yacoto-0.4/yac-menurecordings.c	2012-11-06 00:12:28.418402195 +0100
@@ -110,7 +110,8 @@
    titleBuffer = NULL;
    if( !isDir ) {
       struct tm tm_r;
-      struct tm *t = localtime_r(&(yacRec->CRec()->start), &tm_r);
+      time_t t_rec = yacRec->CRec()->Start();
+      struct tm *t = localtime_r(&t_rec, &tm_r);
       char *s;
 
       int recLength = yacRec->RecLength();
diff -Naur yacoto-0.4_orig/yac-recordings.c yacoto-0.4/yac-recordings.c
--- yacoto-0.4_orig/yac-recordings.c	2010-10-27 21:54:06.000000000 +0200
+++ yacoto-0.4/yac-recordings.c	2012-11-06 00:09:17.870784369 +0100
@@ -195,7 +195,7 @@
       return -1;
 
    for( cMark *mark = marks.First(); mark != 0; mark = marks.Next(mark) ) {
-      int pos = mark->position;
+      int pos = mark->Position();
       if( oldMark >= 0 ) {
          cutLen += pos - oldMark;
          oldMark = -1;
