diff -Naur yacoto-0.4_orig/yacoto/yac_exec.sh yacoto-0.4/yacoto/yac_exec.sh
--- yacoto-0.4_orig/yacoto/yac_exec.sh	2010-10-27 22:05:18.000000000 +0200
+++ yacoto-0.4/yacoto/yac_exec.sh	2011-07-13 01:47:36.357978004 +0200
@@ -133,6 +133,11 @@
    else
       CONV_FILES="$CONV_FILES $VIDEO_DIR/*.ts"
    fi
+   tmp_conv=""
+   for conv_file in $CONV_FILES; do
+      tmp_conv="$tmp_conv \"$conv_file\""
+   done
+   CONV_FILES="$tmp_conv"
    if [ "$DO_TS" = "0" ] ; then
       IDX_FL="$VIDEO_DIR/index.vdr"
       INFO_FL="$VIDEO_DIR/info.vdr"
