diff -Naur yacoto-0.4_orig/yacoto/yac_exec.sh yacoto-0.4/yacoto/yac_exec.sh
--- yacoto-0.4_orig/yacoto/yac_exec.sh	2010-10-27 22:05:18.000000000 +0200
+++ yacoto-0.4/yacoto/yac_exec.sh	2011-10-28 23:29:46.300000009 +0200
@@ -20,6 +20,12 @@
    echo "$(date +"%D %T") - $*" >> $LOG
 }
 
+function parse_videodir() {
+   local videodir="${1//_~_/_:_}"
+   videodir="${videodir%%~*}"
+   echo "${videodir//_:_/_~_}"
+}
+
 [ "$DEBUG" = "1" ] && set -x
 
 YAC_DIR="${YAC_DIR:=/etc/vdr/plugins/yacoto}"
@@ -83,7 +89,7 @@
 
 parm_err=0
 for i in "$@" ; do
-   vid_dir="${i%%~*}"
+   vid_dir="$(parse_videodir ${i})"
    if [ ! -f "$vid_dir/001.vdr" -a ! -f "$vid_dir/00001.ts" ] ; then
       log "<$vid_dir/001.vdr> not found"
       parm_err=1
@@ -102,7 +108,7 @@
 CONV_FILES=""
 
 for i in "$@" ; do
-   VIDEO_DIR=${i%%~*}
+   VIDEO_DIR="$(parse_videodir ${i})"
    if [ -f "$VIDEO_DIR/001.vdr" ] ; then
       DO_TS=0
    else
