diff -Naur yacoto-0.4_orig/yacoto/myvars.conf.sample yacoto-0.4/yacoto/myvars.conf.sample
--- yacoto-0.4_orig/yacoto/myvars.conf.sample	2010-10-27 22:05:18.000000000 +0200
+++ yacoto-0.4/yacoto/myvars.conf.sample	2012-02-17 22:10:58.343999994 +0100
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-SVDRP_CMD="svdrpsend.pl"
+SVDRP_CMD="svdrpsend"
 REMOTECONVERT="1"
 REMOTEIP="192.168.0.2"
 CLIENTIP="192.168.0.28"
diff -Naur yacoto-0.4_orig/yacoto/yac_exec.sh yacoto-0.4/yacoto/yac_exec.sh
--- yacoto-0.4_orig/yacoto/yac_exec.sh	2010-10-27 22:05:18.000000000 +0200
+++ yacoto-0.4/yacoto/yac_exec.sh	2012-02-17 22:10:40.593999995 +0100
@@ -74,7 +74,7 @@
 NICE_LEVEL=${NICE_LEVEL:=10}
 MULTI_AUDIO=${MULTI_AUDIO:=0}
 
-SVDRP_CMD=${SVDRP_CMD:=svdrpsend.pl}
+SVDRP_CMD=${SVDRP_CMD:=svdrpsend}
 
 #set -x
 
diff -Naur yacoto-0.4_orig/yacoto/yac_status.sh yacoto-0.4/yacoto/yac_status.sh
--- yacoto-0.4_orig/yacoto/yac_status.sh	2010-10-27 22:05:18.000000000 +0200
+++ yacoto-0.4/yacoto/yac_status.sh	2012-02-17 22:10:49.793999995 +0100
@@ -1,6 +1,6 @@
 #!/bin/bash
 #set -x
-SVDRP_CMD=${SVDRP_CMD:=svdrpsend.pl}
+SVDRP_CMD=${SVDRP_CMD:=svdrpsend}
 
 function vdr_status() {
    logger -s "vdr_status PLUG bgprocess yacoto PROCESS $STARTTIME $*"
