diff -Naur yacoto-0.4_orig/yacoto/Makefile yacoto-0.4/yacoto/Makefile
--- yacoto-0.4_orig/yacoto/Makefile	2010-10-27 22:05:18.000000000 +0200
+++ yacoto-0.4/yacoto/Makefile	2012-11-06 00:35:59.480761489 +0100
@@ -27,7 +27,7 @@
 HAVE_MP4BOX ?= $(shell which MP4Box &>/dev/null && echo "1")
 HAVE_LAME ?= $(shell which lame &>/dev/null && echo "1")
 HAVE_VORBISTOOLS ?= $(shell which oggenc &>/dev/null && echo "1")
-HAVE_PROJECTX ?= $(shell which projectx_nogui &>/dev/null && echo "1")
+HAVE_PROJECTX ?= $(shell which projectx_cli &>/dev/null && echo "1")
 
 
 #
diff -Naur yacoto-0.4_orig/yacoto/yac_remux.sh yacoto-0.4/yacoto/yac_remux.sh
--- yacoto-0.4_orig/yacoto/yac_remux.sh	2010-10-27 22:05:18.000000000 +0200
+++ yacoto-0.4/yacoto/yac_remux.sh	2012-11-06 00:35:49.815882315 +0100
@@ -4,7 +4,7 @@
 source "${YAC_DIR}/yac_helperfuncs.sh"
 
 set -x
-PROJECTX="/usr/bin/projectx_nogui"
+PROJECTX="/usr/bin/projectx_cli"
 PROJECTX_INI="${YAC_CONF_DIR}/conf/ProjectX.ini"
 
 function usage() {
