diff -Naur yacoto-0.4_orig/yac-menurecordings.c yacoto-0.4/yac-menurecordings.c
--- yacoto-0.4_orig/yac-menurecordings.c	2008-12-18 23:00:29.000000000 +0100
+++ yacoto-0.4/yac-menurecordings.c	2011-09-19 22:47:51.074995000 +0200
@@ -331,7 +331,7 @@
             Skins.Message(mtStatus, cString::sprintf("%s %d: %s", tr("Sorting"), Setup.RecordingsSortMode, RecordingsSortModeTexts[Setup.RecordingsSortMode]));
             return osContinue;
 #elif defined (USE_LIEMIKUUTIO)
-            DirOrderState = !DirOrderState;
+            //DirOrderState = !DirOrderState;
             Set(true);
             return osContinue;
 #else
