diff -Naur gentoo-vdr-scripts-2.0.0_orig/usr/share/vdr/rcscript/pre-start-50-shutdown.sh gentoo-vdr-scripts-2.0.0/usr/share/vdr/rcscript/pre-start-50-shutdown.sh
--- gentoo-vdr-scripts-2.0.0_orig/usr/share/vdr/rcscript/pre-start-50-shutdown.sh	2013-03-30 20:13:40.000000000 +0100
+++ gentoo-vdr-scripts-2.0.0/usr/share/vdr/rcscript/pre-start-50-shutdown.sh	2014-01-06 23:39:21.514000000 +0100
@@ -13,13 +13,13 @@
 	add_param "--shutdown=/usr/share/vdr/bin/vdrshutdown-gate.sh"
 
 	# some sanity warnings
-	if grep -q /usr/share/vdr/bin/vdrshutdown-really.sh /etc/sudoers; then
-		eerror "  depricated entry in /etc/sudoers"
-		eerror "  To keep the shutdown work correctly, remove the line from /etc/sudoers"
-		eerror "  vdr ALL=NOPASSWD:/usr/share/vdr/bin/vdrshutdown-really.sh"
-		eerror "  or call emerge --config gentoo-vdr-scripts"
-		logger -t vdr "ERROR: Depricated entry in /etc/sudoers, please migrate"
-	fi
+#	if grep -q /usr/share/vdr/bin/vdrshutdown-really.sh /etc/sudoers; then
+#		eerror "  depricated entry in /etc/sudoers"
+#		eerror "  To keep the shutdown work correctly, remove the line from /etc/sudoers"
+#		eerror "  vdr ALL=NOPASSWD:/usr/share/vdr/bin/vdrshutdown-really.sh"
+#		eerror "  or call emerge --config gentoo-vdr-scripts"
+#		logger -t vdr "ERROR: Depricated entry in /etc/sudoers, please migrate"
+#	fi
 
 	# warning about depricated acpi wakeup kernel > 2.6.38
 	if [ "${WAKEUP_METHOD}" = acpi ]; then
