diff -Naur upnp_orig/upnp.cpp upnp/upnp.cpp
--- upnp_orig/upnp.cpp	2012-10-28 23:10:33.000000000 +0100
+++ upnp/upnp.cpp	2012-10-28 23:11:37.768380988 +0100
@@ -27,16 +27,14 @@
 const char *cPluginUpnp::CommandLineHelp(void)
 {
   // Return a string that describes all known command line options.
-  std::stringstream ss;
 
-  ss  << "  The UPnP/DLNA server is designed to detect everything automatically.\n"
-      << "  Therefore the user is not required to change most of the settings.\n"
-      << "  \n"
-      << "  -d <DB directory>    --db-dir=<DB directory>  Specifies the directory\n"
-      << "                                                where the the database\n"
-      << "                                                file shall be located.\n";
+  return "  The UPnP/DLNA server is designed to detect everything automatically.\n"
+         "  Therefore the user is not required to change most of the settings.\n"
+         "\n"
+         "  -d <DB directory>    --db-dir=<DB directory>  Specifies the directory\n"
+         "                                                where the the database\n"
+         "                                                file shall be located.\n";
 
-  return ss.str().c_str();
 }
 
 bool cPluginUpnp::ProcessArgs(int argc, char *argv[])
