diff -Naur softhddevice-9999_orig/Makefile softhddevice-9999/Makefile
--- softhddevice-9999_orig/Makefile	2012-11-23 22:39:47.000000000 +0100
+++ softhddevice-9999/Makefile	2012-11-23 22:46:25.974799895 +0100
@@ -18,15 +18,19 @@
 
 ### Configuration (edit this for your needs)
 
+JPEG ?= 1
+OSS ?= 1
+YAEPG ?= 1
+TS_VIDEO ?= 1
+GLX ?= 1
 CONFIG := #-DDEBUG #-DOSD_DEBUG
 CONFIG += -DAV_INFO -DAV_INFO_TIME=3000	# debug a/v sync
-#CONFIG += -DHAVE_PTHREAD_NAME		# supports new pthread_setname_np
+CONFIG += -DHAVE_PTHREAD_NAME		# supports new pthread_setname_np
 #CONFIG += -DNO_TS_AUDIO		# disable ts audio parser
 #CONFIG += -DUSE_TS_VIDEO		# build new ts video parser
 CONFIG += $(shell pkg-config --exists vdpau && echo "-DUSE_VDPAU")
 CONFIG += $(shell pkg-config --exists libva && echo "-DUSE_VAAPI")
 CONFIG += $(shell pkg-config --exists alsa && echo "-DUSE_ALSA")
-CONFIG += -DUSE_OSS
 
 ### The C++ compiler and options:
 
@@ -67,33 +71,56 @@
 DEFINES += $(CONFIG) -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"' \
 	$(if $(GIT_REV), -DGIT_REV='"$(GIT_REV)"')
 
-_CFLAGS = $(DEFINES) $(INCLUDES) \
-	$(shell pkg-config --cflags libavcodec) \
-	`pkg-config --cflags x11 x11-xcb xcb xcb-xv xcb-shm xcb-dpms xcb-atom\
-		xcb-screensaver xcb-randr xcb-glx xcb-icccm xcb-keysyms`\
-	`pkg-config --cflags gl glu` \
-	$(if $(findstring USE_VDPAU,$(CONFIG)), \
-		    `pkg-config --cflags vdpau`) \
-	$(if $(findstring USE_VAAPI,$(CONFIG)), \
-		    `pkg-config --cflags libva-x11 libva-glx libva`) \
-	$(if $(findstring USE_ALSA,$(CONFIG)), \
-		    `pkg-config --cflags alsa`)
+_CFLAGS += $(DEFINES) $(INCLUDES) \
+	$(shell pkg-config --cflags libavcodec x11 x11-xcb xcb xcb-atom xcb-icccm)
+
+ifeq ($(ALSA),1)
+CONFIG += -DUSE_ALSA
+_CFLAGS += $(shell pkg-config --cflags alsa)
+LIBS += $(shell pkg-config --libs alsa)
+endif
+ifeq ($(VDPAU),1)
+CONFIG += -DUSE_VDPAU
+_CFLAGS += $(shell pkg-config --cflags vdpau)
+LIBS += $(shell pkg-config --libs vdpau)
+endif
+ifeq ($(VAAPI),1)
+CONFIG += -DUSE_VAAPI
+_CFLAGS += $(shell pkg-config --cflags libva-x11 libva)
+LIBS += $(shell pkg-config --libs libva-x11 libva)
+endif
+ifeq ($(OSS),1)
+CONFIG += -DUSE_OSS
+endif
+ifeq ($(JPEG),1)
+CONFIG += -DUSE_JPEG
+endif
+ifeq ($(YAEPG),1)
+CONFIG += -DUSE_YAEPG
+endif
+ifeq ($(TS_VIDEO),1)
+CONFIG += -DUSE_TS_VIDEO
+endif
+ifeq ($(GLX),1)
+CONFIG += -DUSE_GLX
+_CFLAGS += $(shell pkg-config --cflags gl glu)
+LIBS += $(shell pkg-config --libs gl glu)
+ifeq ($(VAAPI),1)
+_CFLAGS += $(shell pkg-config --cflags libva-glx)
+LIBS += $(shell pkg-config --libs libva-glx)
+endif
+endif
+ifeq ($(SCREENSAVER),1)
+CONFIG += -DUSE_SCREENSAVER
+_CFLAGS += $(shell pkg-config --cflags xcb-screensaver xcb-dpms)
+LIBS += $(shell pkg-config --libs xcb-screensaver xcb-dpms)
+endif
 
 #override _CFLAGS  += -Werror
 override CXXFLAGS += $(_CFLAGS)
 override CFLAGS	  += $(_CFLAGS)
 
-LIBS += -lrt \
-	$(shell pkg-config --libs libavcodec) \
-	`pkg-config --libs x11 x11-xcb xcb xcb-xv xcb-shm xcb-dpms xcb-atom\
-		xcb-screensaver xcb-randr xcb-glx xcb-icccm xcb-keysyms`\
-	`pkg-config --libs gl glu` \
-	$(if $(findstring USE_VDPAU,$(CONFIG)), \
-		    `pkg-config --libs vdpau`) \
-	$(if $(findstring USE_VAAPI,$(CONFIG)), \
-		    `pkg-config --libs libva-x11 libva-glx libva`) \
-	$(if $(findstring USE_ALSA,$(CONFIG)), \
-		    `pkg-config --libs alsa`)
+LIBS += -lrt $(shell pkg-config --libs libavcodec x11 x11-xcb xcb xcb-icccm)
 
 ### The object files (add further files here):
 
