From adb854352ad77cf6ece91022cd384d52f4e9dbad Mon Sep 17 00:00:00 2001
From: Lucian Muresan <lucianm@users.sourceforge.net>
Date: Fri, 25 Jan 2013 02:43:10 +0100
Subject: [PATCH] vdr tvguide Makefile VDRCONFDIR

---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 7c47228..2fceb21 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(s
 LIBDIR = $(DESTDIR)$(call PKGCFG,libdir)
 LOCDIR = $(DESTDIR)$(call PKGCFG,locdir)
 PLGCFG  = $(call PKGCFG,plgcfg)
-CONFDIR= $(DESTDIR)$(call PKGCFG,configdir)
+VDRCONFDIR= $(DESTDIR)$(call PKGCFG,configdir)
 TMPDIR ?= /tmp
 
 ### The compiler options:
@@ -104,8 +104,8 @@ install-lib: $(SOFILE)
 	install -D $^ $(LIBDIR)/$^.$(APIVERSION)
 
 install-themes:
-	@mkdir -p $(CONFDIR)/themes
-	cp themes/* $(CONFDIR)/themes
+	@mkdir -p $(VDRCONFDIR)/themes
+	cp themes/* $(VDRCONFDIR)/themes
 
 install: install-lib install-i18n install-themes
 
-- 
1.8.1

