diff -ru xine-0.9.0-orig/Makefile xine-0.9.0/Makefile
--- xine-0.9.0-orig/Makefile	2009-01-18 17:40:13.228935882 +0100
+++ xine-0.9.0/Makefile	2009-01-18 17:40:39.315595879 +0100
@@ -37,7 +37,7 @@
 INCLUDES += `pkg-config --cflags libxine` 
 
 # where to create fifos (xine expects them at /tmp/vdr-xine)
-VDR_XINE_FIFO_DIR ?= /tmp/vdr-xine
+VDR_XINE_FIFO_DIR ?= /var/vdr/xine
 
 # can be used to detect inefficient OSD drawing
 # 0 - do not verify whether the dirty area of a bitmap is really dirty
diff -ru xine-0.9.0-orig/xineLib.c xine-0.9.0/xineLib.c
--- xine-0.9.0-orig/xineLib.c	2009-01-18 17:40:13.228935882 +0100
+++ xine-0.9.0/xineLib.c	2009-01-18 17:41:28.815595331 +0100
@@ -2118,7 +2118,7 @@
 
   bool cXineLib::readNoSignalStream(const int index, const string &suffix)
   {
-    string noSignalFileName = m_plugin->ConfigDirectory(PLUGIN_NAME_I18N);
+    string noSignalFileName = "/usr/share/vdr/xine";
 //    noSignalFileName += "/noSignal.pes";
     noSignalFileName += "/noSignal" + suffix + ".mpg";
 
