From 78a4e9e8c79db7dc1ccadf18052cbdede82eec08 Mon Sep 17 00:00:00 2001
From: Lucian Muresan <lucianm@users.sourceforge.net>
Date: Sun, 16 Dec 2012 17:50:59 +0100
Subject: [PATCH] yaepghd-video-scaling-without-YAEPG-vdr-1.7.33

---
 yaepghd.c | 33 ++++++++++++---------------------
 1 file changed, 12 insertions(+), 21 deletions(-)

diff --git a/yaepghd.c b/yaepghd.c
index 868f999..1054616 100644
--- a/yaepghd.c
+++ b/yaepghd.c
@@ -25,12 +25,9 @@
 #include <vdr/device.h>
 #include <vdr/thread.h>
 
-#ifndef YAEPGHDVERSNUM
-#error "You must apply the yaepghd patch for VDR!"
-#endif
 
-#if defined(APIVERSNUM) && APIVERSNUM < 10600
-#error "VDR-1.6.0 API version or greater is required!"
+#if defined(APIVERSNUM) && APIVERSNUM < 10733
+#error "VDR-1.7.33 API version or greater is required!"
 #endif
 
 /**
@@ -2583,11 +2580,13 @@ cYaepghd::Show(void)
 
    /* Set up the video window parameters */
    if (VID_WIN_GEOM.w != 0 && VID_WIN_GEOM.h != 0) {
-      osd->vidWin.x1 = VID_WIN_GEOM.x;
-      osd->vidWin.y1 = VID_WIN_GEOM.y;
-      osd->vidWin.x2 = VID_WIN_GEOM.x + VID_WIN_GEOM.w;
-      osd->vidWin.y2 = VID_WIN_GEOM.y + VID_WIN_GEOM.h;
-      osd->vidWin.bpp = 12;
+      // try to scale if the output device supports it
+      cRect vidWinRect(
+         VID_WIN_GEOM.x,
+         VID_WIN_GEOM.y,
+         VID_WIN_GEOM.w,
+         VID_WIN_GEOM.h);
+      cDevice::PrimaryDevice()->ScaleVideo(cDevice::PrimaryDevice()->CanScaleVideo(vidWinRect));
    }
 
 #ifdef YAEPGHD_REEL_EHD
@@ -2862,17 +2861,10 @@ cYaepghd::SwitchToCurrentChannel(bool closeVidWin)
    const cChannel *gridChan = chanVec[gridEvents->Row()];
 
    if (gridChan && (gridChan->Number() != cDevice::CurrentChannel())) {
-      /*
-       * The "Channel not availaible message" will cause vdr to crash. Do a
-       * "lower level" channel switch to avoid the error message.
-       *
-       * XXX Is this still true ? XXX
-       */
-      eSetChannelResult ret;
 
       /*
        * The eHD card doesn't seem to like changing channels while the video
-       * plane is sacled down.  To get around this problem we close/reopen the
+       * plane is scaled down.  To get around this problem we close/reopen the
        * video window across channel changes.
        */
 #ifdef YAEPGHD_REEL_EHD
@@ -2880,9 +2872,8 @@ cYaepghd::SwitchToCurrentChannel(bool closeVidWin)
       cCondWait::SleepMs(100);
 #endif
 
-      ret = cDevice::PrimaryDevice()->SetChannel(gridChan, true);
-      if (ret != scrOk) {
-         fprintf(stderr, "SetChannel(): %d\n", ret);
+      if (!cDevice::PrimaryDevice()->SwitchChannel(gridChan, true)) {
+         fprintf(stderr, "SwitchChannel(): failed\n");
       }
 
 #ifdef YAEPGHD_REEL_EHD
-- 
1.8.0

