From 1ea38578205535ecca0e76ae24fe4e1da56bb8d9 Mon Sep 17 00:00:00 2001
From: Lucian Muresan <lucianm@users.sourceforge.net>
Date: Thu, 3 Jan 2013 22:10:00 +0100
Subject: [PATCH] yaepghd video scaling without YAEPG vdr-1.7.33 v3

---
 yaepghd.c | 36 +++++++++++++++---------------------
 1 file changed, 15 insertions(+), 21 deletions(-)

diff --git a/yaepghd.c b/yaepghd.c
index 868f999..def53f4 100644
--- a/yaepghd.c
+++ b/yaepghd.c
@@ -25,12 +25,9 @@
 #include <vdr/device.h>
 #include <vdr/thread.h>
 
-#ifndef YAEPGHDVERSNUM
-#error "You must apply the yaepghd patch for VDR!"
-#endif
 
-#if defined(APIVERSNUM) && APIVERSNUM < 10600
-#error "VDR-1.6.0 API version or greater is required!"
+#if defined(APIVERSNUM) && APIVERSNUM < 10733
+#error "VDR-1.7.33 API version or greater is required!"
 #endif
 
 /**
@@ -2476,6 +2473,7 @@ private:
    time_t startTime;
    tArea mainWin;
    cBitmap *mainBmp;
+   cRect videoWindowRect;
 
    std::vector< cChannel * > chanVec;
    const cEvent *event;
@@ -2547,6 +2545,7 @@ cYaepghd::~cYaepghd()
    delete eventDesc;
    delete eventDate;
    delete helpBar;
+   cDevice::PrimaryDevice()->ScaleVideo(); // rescale to full size
 #ifdef YAEPGHD_REEL_EHD
    reelVidWin->Close();
 #endif
@@ -2583,11 +2582,13 @@ cYaepghd::Show(void)
 
    /* Set up the video window parameters */
    if (VID_WIN_GEOM.w != 0 && VID_WIN_GEOM.h != 0) {
-      osd->vidWin.x1 = VID_WIN_GEOM.x;
-      osd->vidWin.y1 = VID_WIN_GEOM.y;
-      osd->vidWin.x2 = VID_WIN_GEOM.x + VID_WIN_GEOM.w;
-      osd->vidWin.y2 = VID_WIN_GEOM.y + VID_WIN_GEOM.h;
-      osd->vidWin.bpp = 12;
+      // ask the output device to scale the video when next flushing the OSD, if it supports this
+      cRect vidWinRect(
+         VID_WIN_GEOM.x,
+         VID_WIN_GEOM.y,
+         VID_WIN_GEOM.w,
+         VID_WIN_GEOM.h);
+      videoWindowRect = cDevice::PrimaryDevice()->CanScaleVideo(vidWinRect);
    }
 
 #ifdef YAEPGHD_REEL_EHD
@@ -2862,17 +2863,10 @@ cYaepghd::SwitchToCurrentChannel(bool closeVidWin)
    const cChannel *gridChan = chanVec[gridEvents->Row()];
 
    if (gridChan && (gridChan->Number() != cDevice::CurrentChannel())) {
-      /*
-       * The "Channel not availaible message" will cause vdr to crash. Do a
-       * "lower level" channel switch to avoid the error message.
-       *
-       * XXX Is this still true ? XXX
-       */
-      eSetChannelResult ret;
 
       /*
        * The eHD card doesn't seem to like changing channels while the video
-       * plane is sacled down.  To get around this problem we close/reopen the
+       * plane is scaled down.  To get around this problem we close/reopen the
        * video window across channel changes.
        */
 #ifdef YAEPGHD_REEL_EHD
@@ -2880,9 +2874,8 @@ cYaepghd::SwitchToCurrentChannel(bool closeVidWin)
       cCondWait::SleepMs(100);
 #endif
 
-      ret = cDevice::PrimaryDevice()->SetChannel(gridChan, true);
-      if (ret != scrOk) {
-         fprintf(stderr, "SetChannel(): %d\n", ret);
+      if (!cDevice::PrimaryDevice()->SwitchChannel(gridChan, true)) {
+         fprintf(stderr, "SwitchChannel(): failed\n");
       }
 
 #ifdef YAEPGHD_REEL_EHD
@@ -2939,6 +2932,7 @@ cYaepghd::Draw(void)
    helpBar->Draw(mainBmp);
 
    osd->DrawBitmap(0, 0, *mainBmp);
+   cDevice::PrimaryDevice()->ScaleVideo(videoWindowRect); // scale to our desired video window size if supported
    osd->Flush();
 }
 
-- 
1.8.0.2

