--- xbmc-9999.ebuild.orig	2013-04-23 04:01:22.000000000 +0200
+++ xbmc-9999.ebuild	2013-05-11 21:16:36.510000000 +0200
@@ -13,7 +13,9 @@
 
 case ${PV} in
 9999)
-	EGIT_REPO_URI="git://github.com/xbmc/xbmc.git"
+	EGIT_REPO_URI="${XBMC_EGIT_REPO_URI:-git://github.com/xbmc/xbmc.git}"
+	EGIT_PROJECT="${PN}${XBMC_EGIT_PROJECT:-}.git"
+	EGIT_BRANCH="${XBMC_EGIT_BRANCH:-master}"
 	inherit git-2
 	SRC_URI="!java? ( mirror://gentoo/${P}-20130413-generated-addons.tar.xz )"
 	;;
@@ -36,7 +38,7 @@
 
 LICENSE="GPL-2"
 SLOT="0"
-IUSE="airplay alsa altivec avahi bluetooth bluray caps cec css debug +fishbmc gles goom java joystick midi mysql neon nfs +opengl profile +projectm pulseaudio pvr +rsxs rtmp +samba +sdl sse sse2 sftp udev upnp +usb vaapi vdpau webserver +X +xrandr"
+IUSE="airplay alsa altivec avahi bluetooth bluray caps cec css debug +fishbmc gles goom java joystick midi mysql neon nfs +opengl profile +projectm pulseaudio pvr +rsxs rtmp +samba +sdl sse sse2 sftp udev upnp +usb vaapi vdpau webserver +X +xrandr ffmpeg"
 REQUIRED_USE="
 	pvr? ( mysql )
 	rsxs? ( X )
@@ -92,8 +94,10 @@
 	media-libs/tiff
 	pulseaudio? ( media-sound/pulseaudio )
 	media-sound/wavpack
-	|| ( media-libs/libpostproc media-video/ffmpeg )
-	>=virtual/ffmpeg-9[encode]
+	ffmpeg? (
+		|| ( media-libs/libpostproc media-video/ffmpeg )
+		>=virtual/ffmpeg-9[encode]
+	)
 	rtmp? ( media-video/rtmpdump )
 	avahi? ( net-dns/avahi )
 	nfs? ( net-fs/libnfs )
@@ -116,7 +120,7 @@
 	vaapi? ( x11-libs/libva[opengl] )
 	vdpau? (
 		|| ( x11-libs/libvdpau >=x11-drivers/nvidia-drivers-180.51 )
-		virtual/ffmpeg[vdpau]
+		ffmpeg? ( virtual/ffmpeg[vdpau] )
 	)
 	X? (
 		x11-apps/xdpyinfo
@@ -210,6 +214,7 @@
 		--disable-optimizations \
 		--enable-external-libraries \
 		--enable-gl \
+		$(use_enable ffmpeg external-ffmpeg) \
 		$(use_enable airplay) \
 		$(use_enable avahi) \
 		$(use_enable bluray libbluray) \
