--- xbmc-9999.ebuild.portage	2014-05-22 22:28:34.210000000 +0200
+++ xbmc-9999.ebuild	2014-05-22 22:26:39.000000000 +0200
@@ -11,10 +11,13 @@
 
 inherit eutils python-single-r1 multiprocessing autotools
 
-CODENAME="Helix"
+CODENAME="Gotham"
+#CODENAME="Helix"
 case ${PV} in
 9999)
-	EGIT_REPO_URI="git://github.com/xbmc/xbmc.git"
+	EGIT_REPO_URI="${XBMC_EGIT_REPO_URI:-git://github.com/xbmc/xbmc.git}"
+	EGIT_PROJECT="${PN}${XBMC_EGIT_PROJECT:-}.git"
+	EGIT_BRANCH="${XBMC_EGIT_BRANCH:-master}"
 	inherit git-2
 	#SRC_URI="!java? ( mirror://gentoo/${P}-20130413-generated-addons.tar.xz )"
 	;;
@@ -225,6 +228,7 @@
 		--disable-ccache \
 		--disable-optimizations \
 		--enable-external-libraries \
+		--enable-external-ffmpeg \
 		$(has_version 'media-video/libav' && echo "--enable-libav-compat") \
 		--enable-gl \
 		$(use_enable airplay) \
