From 79818dec973e55af0ce4b09cd34ff6019321df4b Mon Sep 17 00:00:00 2001
From: Lucian Muresan <lucianm@users.sourceforge.net>
Date: Tue, 8 Jan 2013 00:10:38 +0100
Subject: [PATCH] xmltv2vdr Makefile DESTDIR leftovers epgdata2xmltv

---
 dist/epgdata2xmltv/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dist/epgdata2xmltv/Makefile b/dist/epgdata2xmltv/Makefile
index 3301bd2..8ed8c4d 100644
--- a/dist/epgdata2xmltv/Makefile
+++ b/dist/epgdata2xmltv/Makefile
@@ -67,9 +67,9 @@ install: epgdata2xmltv
 	@mkdir -p $(INSTALL)
 	@install $(STRIP) epgdata2xmltv $(INSTALL)	
 	@mkdir -p $(DESTDIR)/var/lib/epgsources
-	@mkdir -p /var/lib/epgsources/epgdata2xmltv-img
-	@cp epgdata2xmltv.dist /var/lib/epgsources/epgdata2xmltv
-	@chown -R vdr.vdr /var/lib/epgsources
+	@mkdir -p $(DESTDIR)/var/lib/epgsources/epgdata2xmltv-img
+	@cp epgdata2xmltv.dist $(DESTDIR)/var/lib/epgsources/epgdata2xmltv
+	@chown -R vdr.vdr $(DESTDIR)/var/lib/epgsources
 
 dist: clean
 	@-rm -rf $(TMPDIR)/epgdata2xmltv
-- 
1.8.0.2

