diff -Naur cdplayer-1.1.7_orig/Makefile cdplayer-1.1.7/Makefile
--- cdplayer-1.1.7_orig/Makefile	2013-02-12 14:30:42.000000000 +0100
+++ cdplayer-1.1.7/Makefile	2013-02-16 16:53:05.940867000 +0100
@@ -19,8 +19,8 @@
 
 # Use package data if installed...otherwise assume we're under the VDR source directory:
 PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
-LIBDIR = $(DESTDIR)$(call PKGCFG,libdir)
-LOCDIR = $(DESTDIR)$(call PKGCFG,locdir)
+LIBDIR = $(call PKGCFG,libdir)
+LOCDIR = $(call PKGCFG,locdir)
 CFGDIR = $(call PKGCFG,configdir)/plugins/$(PLUGIN)
 PLGCFG = $(call PKGCFG,plgcfg)
 #
