--- vdr-plugin-2.eclass.orig	2013-03-05 10:31:15.000000000 +0100
+++ vdr-plugin-2.eclass	2013-03-10 22:48:45.695145000 +0100
@@ -196,6 +196,21 @@
 	# Use a file instead of a variable as single-stepping via ebuild
 	# destroys environment.
 	touch "${WORKDIR}"/.vdr-plugin_makefile_patched
+
+	# remove inclusion of Make.global
+	sed -i Makefile \
+		-e "s:-include \$(VDRDIR)/Make.global::" \
+		-e "s:include \$(VDRDIR)/Make.global::" \
+		-e "s:-include \$(VDRSRC)/Make.global::" \
+		-e "s:include \$(VDRSRC)/Make.global::" \
+		-e "s:-include \$(VDRSRC)/Make.config::" \
+		-e "s:include \$(VDRSRC)/Make.config::"
+
+	# include Make.config if not included at all (even via PLGCFG)
+	if ! grep -q '\-include \$(VDRDIR)/Make.config' Makefile && ! grep -q '\-include \$(VDRSRC)/Make.config' Makefile && ! grep -q '\-include \$(PLGCFG)' Makefile; then
+		sed -i Makefile \
+			-e "/^### The object files (add further files here)\:/a -include \$(VDRDIR)/Make.config"
+	fi
 }
 
 # Begin new vdr-plugin-2.eclass content
