--- vdr-plugin-2.eclass.orig	2022-09-12 21:42:45.000000000 +0200
+++ vdr-plugin-2.eclass	2022-12-02 01:46:43.473968801 +0100
@@ -27,6 +27,20 @@
 # defaults to ${FILESDIR}/confd. Each config file will be installed as e.g.
 # ${D}/etc/conf.d/vdr.${VDRPLUGIN}
 
+# @ECLASS-VARIABLE: VDR_CONFD_FILE_4ARGSDIR
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# An alternative plugin config file can be specified through the $VDR_CONFD_FILE_4ARGSDIR variable,
+# if using ARGSDIR (>=media-tv/gentoo-vdr-scripts-3.0.0) it defaults to ${FILESDIR}/confd_4argsdir.
+# Each config file will be installed as e.g.
+# ${D}/etc/conf.d/vdr.${VDRPLUGIN}
+
+# @ECLASS-VARIABLE: VDR_ARGS_FILE
+# @DESCRIPTION:
+# A plugin new style argsdir config file can be specified through the $VDR_ARGS_FILE variable, it
+# defaults to ${FILESDIR}/${VDRPLUGIN}.conf. Each config file will be installed as e.g.
+# ${D}/etc/vdr/conf.avail/${VDRPLUGIN}.conf
+
 # @ECLASS_VARIABLE: VDR_RCADDON_FILE
 # @DEFAULT_UNSET
 # @DESCRIPTION:
@@ -36,6 +50,16 @@
 # ${FILESDIR}/rc-addon.sh and will be installed as
 # ${VDR_RC_DIR}/plugin-${VDRPLUGIN}.sh
 #
+# @ECLASS-VARIABLE: VDR_RCADDON_FILE_4ARGSDIR
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Installing rc-addon files is basically the same as for plugin config files
+# (see above), it's just using the $VDR_RCADDON_FILE_4ARGSDIR variable instead.
+# The default value when $VDR_RCADDON_FILE_4ARGSDIR is undefined and if using ARGSDIR
+# (>=media-tv/gentoo-vdr-scripts-3.0.0) is:
+# ${FILESDIR}/rc-addon_4argsdir.sh and will be installed as
+# ${VDR_RC_DIR}/plugin-${VDRPLUGIN}.sh
+#
 # The rc-addon files will be sourced by the startscript when the specific plugin
 # has been enabled.
 # rc-addon files may be used to prepare everything that is necessary for the
@@ -635,25 +659,49 @@
 			done
 	fi
 
+	local confd_name="confd"
+	if has_version ">=media-tv/gentoo-vdr-scripts-3.0_rc2"; then
+		confd_name="confd_4argsdir"
+		[[ -n ${VDR_CONFD_FILE_4ARGSDIR} ]] && [[ -e ${VDR_CONFD_FILE_4ARGSDIR} ]] && VDR_CONFD_FILE=${VDR_CONFD_FILE_4ARGSDIR}
+	fi
 	# if VDR_CONFD_FILE is empty and ${FILESDIR}/confd exists take it
-	[[ -z ${VDR_CONFD_FILE} ]] && [[ -e ${FILESDIR}/confd ]] && VDR_CONFD_FILE=${FILESDIR}/confd
+	[[ -z ${VDR_CONFD_FILE} ]] && [[ -e ${FILESDIR}/${confd_name} ]] && VDR_CONFD_FILE=${FILESDIR}/${confd_name}
 
 	if [[ -n ${VDR_CONFD_FILE} ]]; then
 		newconfd "${VDR_CONFD_FILE}" vdr.${VDRPLUGIN}
 	fi
 
+	local rcaddon_name="rc-addon.sh"
+	if has_version ">=media-tv/gentoo-vdr-scripts-3.0_rc2"; then
+		confd_name="rc-addon_4argsdir.sh"
+		[[ -n ${VDR_RCADDON_FILE_4ARGSDIR} ]] && [[ -e ${VDR_RCADDON_FILE_4ARGSDIR} ]] && VDR_RCADDON_FILE=${VDR_RCADDON_FILE_4ARGSDIR}
+	fi
 	# if VDR_RCADDON_FILE is empty and ${FILESDIR}/rc-addon.sh exists take it
-	[[ -z ${VDR_RCADDON_FILE} ]] && [[ -e ${FILESDIR}/rc-addon.sh ]] && VDR_RCADDON_FILE=${FILESDIR}/rc-addon.sh
+	[[ -z ${VDR_RCADDON_FILE} ]] && [[ -e ${FILESDIR}/${rcaddon_name} ]] && VDR_RCADDON_FILE=${FILESDIR}/${rcaddon_name}
 
 	if [[ -n ${VDR_RCADDON_FILE} ]]; then
 		insinto "${VDR_RC_DIR}"
 		newins "${VDR_RCADDON_FILE}" plugin-${VDRPLUGIN}.sh
 	fi
+
+	# if VDR_ARGS_FILE is empty and ${FILESDIR}/argsfile exists take it
+	[[ -z ${VDR_ARGS_FILE} ]] && [[ -e ${FILESDIR}/argsfile ]] && VDR_ARGS_FILE=${FILESDIR}/argsfile
+
+	if [[ -n ${VDR_ARGS_FILE} ]]; then
+		insinto "/etc/vdr/conf.avail"
+		newins "${VDR_ARGS_FILE}" ${VDRPLUGIN}.conf
+	fi
 }
 
 vdr-plugin-2_pkg_postinst() {
 	vdr-plugin-2_print_enable_command
 
+	if [[ -n "${VDR_ARGS_FILE}" ]]; then
+		elog "Please have a look at the argsdir-file"
+		elog "\t/etc/vdr/conf.avail/${VDRPLUGIN}.conf"
+		elog
+	fi
+
 	if [[ -n "${VDR_CONFD_FILE}" ]]; then
 		elog "Please have a look at the config-file"
 		elog "\t/etc/conf.d/vdr.${VDRPLUGIN}"
