--- recording.c	2016/12/22 12:58:20	4.6
+++ recording.c	2017/01/01 17:52:51
@@ -1914,16 +1914,20 @@
 void cRecordingsHandler::Action(void)
 {
   while (Running()) {
+        bool Sleep = false;
         {
           cMutexLock MutexLock(&mutex);
-          while (cRecordingsHandlerEntry *r = operations.First()) {
-                if (!r->Active(error))
-                   operations.Del(r);
-                }
-          if (!operations.Count())
+          if (cRecordingsHandlerEntry *r = operations.First()) {
+             if (!r->Active(error))
+                operations.Del(r);
+             else
+                Sleep = true;
+             }
+          else
              break;
         }
-        cCondWait::SleepMs(100);
+        if (Sleep)
+           cCondWait::SleepMs(100);
         }
 }
 
