From 752cd0ad3ddd1e761b5386633995238b55033b74 Mon Sep 17 00:00:00 2001
From: Lucian Muresan <lucianm@users.sourceforge.net>
Date: Sun, 18 Jan 2015 23:30:18 +0100
Subject: [PATCH 1/2] Backport of systemd support from vdr-2.1.6 and vdr-2.1.7

---
 HISTORY     | 5 +++++
 Makefile    | 5 +++++
 interface.c | 6 ++++++
 vdr.c       | 7 +++++++
 4 files changed, 23 insertions(+)

diff --git a/HISTORY b/HISTORY
index fa6a362..2b00147 100644
--- a/HISTORY
+++ b/HISTORY
@@ -7915,3 +7915,8 @@ Video Disk Recorder Revision History
   on a system with a large number of recordings.
 - The APIVERSION has been increased to 2.0.6 due to the changes to pat.h, sdt.h and
   the functional modification to cFont::CreateFont().
+
+2015-01-18: Version 2.0.6 + backport of systemd support from versions 2.1.6 and 2.1.7
+
+- Added support for systemd (thanks to Christopher Reimer). To activate this you
+  need to add "SDNOTIFY=1" to the 'make' call.
diff --git a/Makefile b/Makefile
index d24d08f..b5c1b2e 100644
--- a/Makefile
+++ b/Makefile
@@ -93,6 +93,11 @@ INCLUDES += $(shell pkg-config --cflags fribidi)
 DEFINES += -DBIDI
 LIBS += $(shell pkg-config --libs fribidi)
 endif
+ifdef SDNOTIFY
+INCLUDES += $(shell pkg-config --cflags libsystemd-daemon)
+DEFINES += -DSDNOTIFY
+LIBS += $(shell pkg-config --libs libsystemd-daemon)
+endif
 
 LIRC_DEVICE ?= /var/run/lirc/lircd
 
diff --git a/interface.c b/interface.c
index 82091d9..93a78d2 100644
--- a/interface.c
+++ b/interface.c
@@ -10,6 +10,9 @@
 #include "interface.h"
 #include <ctype.h>
 #include <stdlib.h>
+#ifdef SDNOTIFY
+#include <systemd/sd-daemon.h>
+#endif
 #include <unistd.h>
 #include "i18n.h"
 #include "status.h"
@@ -159,6 +162,9 @@ void cInterface::LearnKeys(void)
       bool known = Keys.KnowsRemote(Remote->Name());
       dsyslog("remote control %s - %s", Remote->Name(), known ? "keys known" : "learning keys");
       if (!known) {
+#ifdef SDNOTIFY
+         sd_notify(0, "READY=1\nSTATUS=Learning keys...");
+#endif
          cSkinDisplayMenu *DisplayMenu = Skins.Current()->DisplayMenu();
          DisplayMenu->SetMenuCategory(mcUnknown);
          char Headline[256];
diff --git a/vdr.c b/vdr.c
index ffd0e01..224e935 100644
--- a/vdr.c
+++ b/vdr.c
@@ -34,6 +34,9 @@
 #include <stdlib.h>
 #include <sys/capability.h>
 #include <sys/prctl.h>
+#ifdef SDNOTIFY
+#include <systemd/sd-daemon.h>
+#endif
 #include <termios.h>
 #include <unistd.h>
 #include "audio.h"
@@ -837,6 +840,10 @@ int main(int argc, char *argv[])
      alarm(WatchdogTimeout); // Initial watchdog timer start
      }
 
+#ifdef SDNOTIFY
+  sd_notify(0, "READY=1\nSTATUS=Ready");
+#endif
+
   // Main program loop:
 
 #define DELETE_MENU ((IsInfoMenu &= (Menu == NULL)), delete Menu, Menu = NULL)
-- 
1.9.4.msysgit.0

