From eafc01b42f50632c98e5f871b491b9124427726c Mon Sep 17 00:00:00 2001
From: Lucian Muresan <lucianm@users.sourceforge.net>
Date: Sun, 15 Feb 2015 10:52:45 +0100
Subject: [PATCH] generate arg file templates to /tmp

---
 vdr.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/vdr.c b/vdr.c
index 1436c3e..4afe662 100644
--- a/vdr.c
+++ b/vdr.c
@@ -185,6 +185,7 @@ const char *VideoDirectory = DEFAULTVIDEODIR;
 
 #define DEFAULTCONFDIR dd(CONFDIR, VideoDirectory)
 #define DEFAULTARGSDIR dd(ARGSDIR, "/etc/vdr/conf.d")
+#define DEFAULTTPLARGDIR "/tmp"
 #define DEFAULTCACHEDIR dd(CACHEDIR, VideoDirectory)
 #define DEFAULTRESDIR dd(RESDIR, ConfigDirectory)
 #define DEFAULTPLUGINDIR PLUGINDIR
@@ -282,7 +283,7 @@ const char *CommandLineHelp(void)
     DEFAULTSVDRPPORT,
     DEFAULTRESDIR,
     DEFAULTARGSDIR,
-    DEFAULTARGSDIR,
+    DEFAULTTPLARGDIR,
     DEFAULTVIDEODIR,
     DEFAULTWATCHDOG));
     return sCliHelp;
@@ -598,7 +599,7 @@ int main(int argc, char *argv[])
                     }
           case 't' | 0x100:
                     GenerateConfTemplates = true;
-                    ArgTemplatesDirectory = optarg ? optarg : DEFAULTARGSDIR;
+                    ArgTemplatesDirectory = optarg ? optarg : DEFAULTTPLARGDIR;
                     break;
           case 't': Terminal = optarg;
                     if (access(Terminal, R_OK | W_OK) < 0) {
-- 
2.2.2

