diff -Naur vdr-2.4.1_orig/PLUGINS/src/skincurses/skincurses.c vdr-2.4.1/PLUGINS/src/skincurses/skincurses.c
--- vdr-2.4.1_orig/PLUGINS/src/skincurses/skincurses.c	2019-03-12 13:28:04.000000000 +0100
+++ skincurses/skincurses.c	2020-05-08 21:06:13.388558987 +0200
@@ -83,7 +83,7 @@
   start_color();
   leaveok(stdscr, true);
 
-  window = subwin(stdscr, ScOsdHeight, ScOsdWidth, 0, 0);
+  window = stdscr;
   syncok(window, true);
 }
 
@@ -92,7 +92,6 @@
   if (window) {
      werase(window);
      Flush();
-     delwin(window);
      window = NULL;
      }
 }
@@ -122,17 +121,14 @@
      savedRegion = NULL;
      }
   savedRegion = newwin(y2 - y1 + 1, x2 - x1 + 1, y1, x1);
-  copywin(window, savedRegion, y1, x1, 0, 0, y2 - y1, x2 - x1, false);
+  if (savedRegion)
+     copywin(window, savedRegion, y1, x1, 0, 0, y2 - y1, x2 - x1, false);
 }
 
 void cCursesOsd::RestoreRegion(void)
 {
-  int begy, begx;
-  int maxy, maxx;
-  getmaxyx(savedRegion, maxy,maxx);
-  getbegyx(savedRegion, begy,begx);
   if (savedRegion) {
-     copywin(savedRegion, window, 0, 0, begy, begx, maxy - begy, maxx - begx, false);
+     overwrite(savedRegion, window);
      delwin(savedRegion);
      savedRegion = NULL;
      }
@@ -837,8 +833,8 @@
   getmaxyx(w, maxy,maxx);
   getbegyx(w, begy,begx);
   if (w) {
-     ScOsdWidth  = maxx - begx + 1;
-     ScOsdHeight = maxy - begy + 1;
+     ScOsdWidth  = maxx - begx;
+     ScOsdHeight = maxy - begy;
      return true;
      }
   return false;
