diff -Naur w_scan_cpp-20220105_orig/Makefile w_scan_cpp-20220105/Makefile
--- w_scan_cpp-20220105_orig/Makefile	2022-01-05 10:56:40.000000000 +0100
+++ w_scan_cpp-20220105/Makefile	2022-12-05 03:13:50.318324158 +0100
@@ -217,7 +217,7 @@
 
 
 SOURCES           := $(wildcard $(srcdir)/*.cpp)
-VDR_SOURCES       := $(shell find $(vdrdir)  -maxdepth 1 ! -name "vdr.c" -name "*.c" 2>/dev/null)
+VDR_SOURCES       := $(wildcard $(vdrdir)/*.c)
 LIBSI_SOURCES     := $(wildcard $(vdrlibsidir)/*.c)
 WIRBELSCAN_SOURCES = $(wildcard $(pluginsrcdir)/wirbelscan/*.cpp)
 SATIP_SOURCES      = $(wildcard $(pluginsrcdir)/satip/*.c)
