From 2f644231b222e117afae95ab49a771e61d96caba Mon Sep 17 00:00:00 2001
From: Lucian Muresan <lucianm@users.sourceforge.net>
Date: Mon, 30 Jun 2014 18:46:57 +0200
Subject: [PATCH] pluginsdir_systemd_service

---
 contrib/epgd.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/epgd.service b/contrib/epgd.service
index 6122cc4..9708111 100644
--- a/contrib/epgd.service
+++ b/contrib/epgd.service
@@ -4,7 +4,7 @@ After=mysqld.service
 
 [Service]
 Type=forking
-ExecStart=/usr/bin/epgd -c /etc/epgd
+ExecStart=/usr/bin/epgd -c /etc/epgd -p /usr/lib/epgd/plugins
 
 [Install]
 WantedBy=multi-user.target
-- 
1.9.3

