diff -Naur upnp_orig/media/mediaManager.cpp upnp/media/mediaManager.cpp
--- upnp_orig/media/mediaManager.cpp	2012-10-27 22:46:25.000000000 +0200
+++ upnp/media/mediaManager.cpp	2012-10-27 22:46:55.756637548 +0200
@@ -475,7 +475,7 @@
 
     connection = tntdb::connect(ss.str());
 
-    dsyslog("UPNP\tPreparing database structure...");
+    dsyslog("UPnP\tPreparing database structure...");
 
     if(!CheckIntegrity()){
       connection.beginTransaction();
@@ -588,7 +588,7 @@
     return false;
   }
 
-  dsyslog("UPNP\tLoading Plugins...");
+  dsyslog("UPnP\tLoading Plugins...");
   pluginManager = new upnp::cPluginManager();
 
   if(!pluginManager->LoadPlugins(pluginDirectory)){
diff -Naur upnp_orig/media/pluginManager.cpp upnp/media/pluginManager.cpp
--- upnp_orig/media/pluginManager.cpp	2012-10-27 22:46:25.000000000 +0200
+++ upnp/media/pluginManager.cpp	2012-10-27 22:48:37.568637548 +0200
@@ -405,7 +405,7 @@
       return true;
     }
   } else {
-    esyslog("UPnP\tError while opening plugin: %s", error);
+    esyslog("UPnP\tError while opening plugin: '%s', %s", file.c_str(), error);
   }
 
   return false;
diff -Naur upnp_orig/server/server.cpp upnp/server/server.cpp
--- upnp_orig/server/server.cpp	2012-10-27 22:46:25.000000000 +0200
+++ upnp/server/server.cpp	2012-10-27 22:49:43.095637547 +0200
@@ -270,7 +270,7 @@
 
 void cMediaServer::RegisterService(cUPnPService* service){
   if(service != NULL){
-    cout << "Registered service: " << service->GetServiceDescription().serviceType << endl;
+    dsyslog("UPnP\tRegistered service: %s", service->GetServiceDescription().serviceType.c_str());
     GetServices()[service->GetServiceDescription().serviceID] = service;
   }
 }
