diff -Naur control-0.0.2a_orig/Makefile control-0.0.2a/Makefile
--- control-0.0.2a_orig/Makefile	2004-07-02 21:24:22.000000000 +0200
+++ control-0.0.2a/Makefile	2010-03-05 12:26:09.000000000 +0100
@@ -16,7 +16,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual -fPIC
 
 ### The directory environment:
 
diff -Naur control-0.0.2a_orig/formatter.c control-0.0.2a/formatter.c
--- control-0.0.2a_orig/formatter.c	2004-07-02 21:24:22.000000000 +0200
+++ control-0.0.2a/formatter.c	2010-03-05 12:16:36.000000000 +0100
@@ -165,7 +165,7 @@
 
   if (Title) {
 
-    char* t = strchr(Title, '\t');
+    const char* t = strchr(Title, '\t');
     if (t) {
 
       int size = t - Title;
@@ -300,7 +300,7 @@
 
   for (;;) {
 
-    char* t = strchr(text, '\t');
+    const char* t = strchr(text, '\t');
     if (t) {
       // Tab found - check if the tab can be used
 
