diff -ru control-0.0.2a-orig/gateway.c control-0.0.2a/gateway.c
--- control-0.0.2a-orig/gateway.c	2006-12-03 16:44:46.170845357 +0100
+++ control-0.0.2a/gateway.c	2006-12-03 16:44:58.149701387 +0100
@@ -100,7 +100,7 @@
 
     if (_stack.poll(100)) {
 
-      uint64 Command = 0;
+      uint64_t Command = 0;
       uint i = 0;
       cTimeMs t0;
       t0.Set();
diff -ru control-0.0.2a-orig/keyboard.c control-0.0.2a/keyboard.c
--- control-0.0.2a-orig/keyboard.c	2006-12-03 16:44:46.170845357 +0100
+++ control-0.0.2a/keyboard.c	2006-12-03 16:45:29.825253136 +0100
@@ -18,7 +18,7 @@
 
 struct tKbdMap {
   eKbdFunc func;
-  uint64 code;
+  uint64_t code;
 };
 
 
@@ -117,7 +117,7 @@
 
 
 
-bool cCtrlKeyboard::intPut(uint64 Command, bool Repeat, bool Release) {
+bool cCtrlKeyboard::intPut(uint64_t Command, bool Repeat, bool Release) {
 
   bool ret = cRemote::Put(Command, Repeat, Release);
 conLOG_KEYB("%s cKbdRemoteForControl::intPut(0x%016LX)", ret ? "true" : "false", Command);
@@ -126,7 +126,7 @@
 
 
 
-bool cCtrlKeyboard::Put(uint64 Command, bool Repeat, bool Release) {
+bool cCtrlKeyboard::Put(uint64_t Command, bool Repeat, bool Release) {
 
   if (! _pRemote)
     _pRemote = new cCtrlKeyboard();
@@ -136,7 +136,7 @@
 
 
 
-int cCtrlKeyboard::MapCodeToFunc(uint64 Code) {
+int cCtrlKeyboard::MapCodeToFunc(uint64_t Code) {
 
   for (const tKbdMap *p = keyboardKbdMap; p->func != kfNone; p++) {
     if (p->code == Code)
diff -ru control-0.0.2a-orig/keyboard.h control-0.0.2a/keyboard.h
--- control-0.0.2a-orig/keyboard.h	2006-12-03 16:44:46.170845357 +0100
+++ control-0.0.2a/keyboard.h	2006-12-03 16:45:14.261542675 +0100
@@ -30,11 +30,11 @@
   cCtrlKeyboard();
   virtual ~cCtrlKeyboard();
 
-  bool intPut(uint64 Command, bool Repeat, bool Release);
+  bool intPut(uint64_t Command, bool Repeat, bool Release);
 
 public:
-  static bool Put(uint64 Command, bool Repeat = false, bool Release = false);
-  static int MapCodeToFunc(uint64 Code);
+  static bool Put(uint64_t Command, bool Repeat = false, bool Release = false);
+  static int MapCodeToFunc(uint64_t Code);
 };
 
 
