diff -Naur control-0.0.2a-kw3_orig/gateways.h control-0.0.2a-kw3/gateways.h
--- control-0.0.2a-kw3_orig/gateways.h	2004-07-02 21:24:22.000000000 +0200
+++ control-0.0.2a-kw3/gateways.h	2018-12-24 03:13:22.789179578 +0100
@@ -9,7 +9,7 @@
 #ifndef _CTRLGATEWAYS_H_
 #define _CTRLGATEWAYS_H_
 
-#include <vdr/svdrp.h>  // we need the socket class
+#include "vdr-socket.h"
 #include <vdr/thread.h>
 
 
diff -Naur control-0.0.2a-kw3_orig/Makefile control-0.0.2a-kw3/Makefile
--- control-0.0.2a-kw3_orig/Makefile	2006-04-23 17:28:37.000000000 +0200
+++ control-0.0.2a-kw3/Makefile	2018-12-24 03:13:45.641709516 +0100
@@ -46,7 +46,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o gateways.o gateway.o state.o formatter.o StringArray.o keyboard.o stack.o telnet.o
+OBJS = $(PLUGIN).o vdr-socket.o gateways.o gateway.o state.o formatter.o StringArray.o keyboard.o stack.o telnet.o
 
 ### Targets:
 
diff -Naur control-0.0.2a-kw3_orig/vdr-socket.c control-0.0.2a-kw3/vdr-socket.c
--- control-0.0.2a-kw3_orig/vdr-socket.c	1970-01-01 01:00:00.000000000 +0100
+++ control-0.0.2a-kw3/vdr-socket.c	2018-12-24 03:11:15.000000000 +0100
@@ -0,0 +1,114 @@
+#include "vdr-socket.h"
+
+#include <arpa/inet.h>
+//#include <ctype.h>
+//#include <errno.h>
+//#include <fcntl.h>
+#include <ifaddrs.h>
+#include <netinet/in.h>
+//#include <stdarg.h>
+//#include <stdio.h>
+//#include <stdlib.h>
+//#include <string.h>
+//#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <vdr/tools.h>
+#include <vdr/config.h>
+//#include "device.h"
+//#include "eitscan.h"
+//#include "keys.h"
+//#include "menu.h"
+//#include "plugin.h"
+//#include "recording.h"
+//#include "remote.h"
+//#include "skins.h"
+//#include "timers.h"
+//#include "videodir.h"
+
+cSocket::cSocket(int Port, int Queue)
+{
+  port = Port;
+  sock = -1;
+  queue = Queue;
+}
+
+cSocket::~cSocket()
+{
+  Close();
+}
+
+void cSocket::Close(void)
+{
+  if (sock >= 0) {
+     close(sock);
+     sock = -1;
+     }
+}
+
+bool cSocket::Open(void)
+{
+  if (sock < 0) {
+     // create socket:
+     sock = socket(PF_INET, SOCK_STREAM, 0);
+     if (sock < 0) {
+        LOG_ERROR;
+        port = 0;
+        return false;
+        }
+     // allow it to always reuse the same port:
+     int ReUseAddr = 1;
+     setsockopt(sock, SOL_SOCKET, SO_REUSEADDR, &ReUseAddr, sizeof(ReUseAddr));
+     //
+     struct sockaddr_in name;
+     name.sin_family = AF_INET;
+     name.sin_port = htons(port);
+     name.sin_addr.s_addr = SVDRPhosts.LocalhostOnly() ? htonl(INADDR_LOOPBACK) : htonl(INADDR_ANY);
+     if (bind(sock, (struct sockaddr *)&name, sizeof(name)) < 0) {
+        LOG_ERROR;
+        Close();
+        return false;
+        }
+     // make it non-blocking:
+     int oldflags = fcntl(sock, F_GETFL, 0);
+     if (oldflags < 0) {
+        LOG_ERROR;
+        return false;
+        }
+     oldflags |= O_NONBLOCK;
+     if (fcntl(sock, F_SETFL, oldflags) < 0) {
+        LOG_ERROR;
+        return false;
+        }
+     // listen to the socket:
+     if (listen(sock, queue) < 0) {
+        LOG_ERROR;
+        return false;
+        }
+     }
+  return true;
+}
+
+int cSocket::Accept(void)
+{
+  if (Open()) {
+     struct sockaddr_in clientname;
+     uint size = sizeof(clientname);
+     int newsock = accept(sock, (struct sockaddr *)&clientname, &size);
+     if (newsock > 0) {
+        bool accepted = SVDRPhosts.Acceptable(clientname.sin_addr.s_addr);
+        if (!accepted) {
+           const char *s = "Access denied!\n";
+           if (write(newsock, s, strlen(s)) < 0)
+              LOG_ERROR;
+           close(newsock);
+           newsock = -1;
+           }
+        isyslog("connect from %s, port %hu - %s", inet_ntoa(clientname.sin_addr), ntohs(clientname.sin_port), accepted ? "accepted" : "DENIED");
+        }
+     else if (errno != EINTR && errno != EAGAIN)
+        LOG_ERROR;
+     return newsock;
+     }
+  return -1;
+}
diff -Naur control-0.0.2a-kw3_orig/vdr-socket.h control-0.0.2a-kw3/vdr-socket.h
--- control-0.0.2a-kw3_orig/vdr-socket.h	1970-01-01 01:00:00.000000000 +0100
+++ control-0.0.2a-kw3/vdr-socket.h	2018-12-24 03:10:11.000000000 +0100
@@ -0,0 +1,17 @@
+#ifndef __VDRSOCKET_H
+#define __VDRSOCKET_H
+
+class cSocket {
+private:
+  int port;
+  int sock;
+  int queue;
+  void Close(void);
+public:
+  cSocket(int Port, int Queue = 1);
+  ~cSocket();
+  bool Open(void);
+  int Accept(void);
+  };
+
+#endif //__VDRSOCKET_H
